docs/docs/docs/01-core/utils/async.md

```diff
@@ -12,17 +12,64 @@ Async utils are grouping all function that interact with async stuff like Promis
 
 ## Detailed design
 
-Available functions:
+### mapAsync
 
-- pipeAsync
-- mapAsync
-- reduceAsync
+The `mapAsync` function is an asynchronous version of the `Array.prototype.map` method.
 
-[See API reference](../../../api/api.mdx) (TODO)
+Example usage:
+
+```js
+const input = [1, 2, 3];
+
+const output = await mapAsync(input, async (item) => {
+  return item * 2;
+});
+
+console.log(output); // [2, 4, 6]
+```
+
+### reduceAsync
+
+The `reduceAsync` function is an asynchronous version of the `Array.prototype.reduce` method.
+
+Example usage:
+
+```js
+const input = [1, 2, 3];
+
+const reducer = reduceAsync(input);
+const output = await reducer(async (accumulator, item) => {
+  return accumulator + item;
+}, 0);
```

Comment:
I wonder why did we decide to make the 'input' to be the first argument of the async reducer, and not the callbackFunction and initial value. @nathan-pichon 
Same for the map async, it could be easier to pipe into other places.

Reply:
I do not remember well.
I know it was a huge topic when I implemented it.
I think I never really found a perfect solution and it may be better to reverse the parameters to have `accumulator, defaultValue, function`. Also, the initial value is optional, so it should be `(accumulator, defaultValue) => (array) => {}` I guess 🤔

Reply:
Also, I think those example should be in the API ref, not in the docs

Reply:
Makes sense
> Also, I think those example should be in the API ref, not in the docs
imo it's easy to spot them here as its the first place someone would look into. I don't see much value having this file if there are no examples. In which file you would put it?

Reply:
do you think we could move this file into api ref directly?

Reply:
That's right, we need full new pages for Strapi utils in the API ref 🤔 
That looks like a lot of work. Maybe we can plan it and keep this file for now and move it when we do the utils API ref doc?

Reply:
*I think it's still relevant to have both files. the "When to use", "Should I add a function here" and "Potential improvements" are well placed here.

Reply:
I would be ok, but probably this could be a good oportunity to start the work on the API ref too.